YaleMobile/Branch View Controllers/Laundry/YMLaundryDetailViewController.m

```diff
@@ -54,14 +56,30 @@ - (void)viewDidLoad
 
 - (void)viewDidAppear:(BOOL)animated
 {
+  // implement a loop here so the table view reloads itself every minute. This way the timers automatically update when there should be a change. This assumes that a washer is running, so its timer is counting down. Possibly could implement a check to see if a washer/dryer is actually running before setting up this loop.
+  // make sure to stop all selector loops when this ViewController goes away.
+  [self performSelector:@selector(refreshEveryMinute) withObject:nil afterDelay:AUTO_LAUNDRY_REFRESH_RATE];
   [super viewDidAppear:animated];
 }
 
+- (void)viewWillDisappear:(BOOL)animated
+{
+  // stop the refresh loop
+  [NSObject cancelPreviousPerformRequestsWithTarget:self];
+  [super viewWillDisappear:animated];
+}
+
 - (void)back:(id)sender
 {
   [self.navigationController popViewControllerAnimated:YES];
 }
 
+- (void)refreshEveryMinute
+{
+  [self refresh];
+  [self performSelector:@selector(refreshEveryMinute) withObject:nil afterDelay:AUTO_LAUNDRY_REFRESH_RATE];
```

Comment:
Hi Dan,

Thanks for the tip. I'm new to collaborating over github. Should I make the change and make a new pull request?

Lee Danilek

> On Jan 30, 2015, at 8:52 PM, Danqing Liu <notifications@github.com> wrote:
> 
> In YaleMobile/Branch View Controllers/Laundry/YMLaundryDetailViewController.m:
> 
> >  - (void)back:(id)sender
> >  {
> >    [self.navigationController popViewControllerAnimated:YES];
> >  }
> >  
> > +- (void)refreshEveryMinute
> > +{
> > +  [self refresh];
> > +  [self performSelector:@selector(refreshEveryMinute) withObject:nil afterDelay:AUTO_LAUNDRY_REFRESH_RATE];
> This probably should be moved to refresh success. Otherwise if refresh fails this will go on and on and keep failing. If you do this in refresh success you probably don't need to call it in viewDidAppear either.
> 
> —
> Reply to this email directly or view it on GitHub.
>